File: AssetInsight/src/main/java/br/unipar/assetinsight/dtos/responses/OrdemServicoResponse.java
package br.unipar.assetinsight.dtos.responses;

import br.unipar.assetinsight.enums.StatusOrdemServicoEnum;

import java.io.Serializable;
import java.sql.Timestamp;
import java.util.List;

/**
 * DTO for {@link br.unipar.assetinsight.entities.OrdemServicoEntity}
 */
public record OrdemServicoResponse(
        long id,
        String descricao,
        Timestamp data,
        List<ServicoRespose> servicos,
        double valorTotal,
        StatusOrdemServicoEnum status,
        UsuarioResponse lastChangedBy,
        Timestamp lastChange,
        boolean arquivado
) implements Serializable { }

File: Frontend/src/app/dtos/responses/OrdemServico.response.ts
import {ServicoResponse} from "./Servico.response";
import {UsuarioResponse} from "./Usuario.response";
import {StatusOrdem} from "../enums/StatusOrdem.enum";

export interface OrdemServicoResponse {
  id: Number;
  descricao: String;
  data: Date;
  servicos: ServicoResponse[];
  qtdServicos: Number;
  valorTotal: Number;
  status: StatusOrdem;
  lastChangedBy: UsuarioResponse;
  lastChange: Date;
  arquivada: Boolean;
}




